stage1/init/kvm/network.go

```diff
@@ -33,36 +40,121 @@ func GetNetworkDescriptions(n *networking.Networking) []netDescriber {
 
 // netDescriber is something that describes network configuration
 type netDescriber interface {
-	HostIP() net.IP
 	GuestIP() net.IP
 	Mask() net.IP
 	IfName() string
 	IPMasq() bool
+	Name() string
+	Gateway() net.IP
+	Routes() []types.Route
 }
 
-// GetKVMNetArgs returns additional arguments that need to be passed to kernel
-// and lkvm tool to configure networks properly.
+// GetKVMNetArgs returns additional arguments that need to be passed
+// to lkvm tool to configure networks properly.
 // Logic is based on Network configuration extracted from Networking struct
 // and essentially from activeNets that expose netDescriber behavior
-func GetKVMNetArgs(nds []netDescriber) ([]string, []string, error) {
+func GetKVMNetArgs(nds []netDescriber) ([]string, error) {
 
 	lkvmArgs := []string{}
-	kernelParams := []string{}
-
-	for i, nd := range nds {
-		// https://www.kernel.org/doc/Documentation/filesystems/nfs/nfsroot.txt
-		// ip=<client-ip>:<server-ip>:<gw-ip>:<netmask>:<hostname>:<device>:<autoconf>:<dns0-ip>:<dns1-ip>
-		var gw string
-		if nd.IPMasq() {
-			gw = nd.HostIP().String()
-		}
-		kernelParam := fmt.Sprintf("ip=%s::%s:%s::%s:::", nd.GuestIP(), gw, nd.Mask(), fmt.Sprintf(networking.IfNamePattern, i))
-		kernelParams = append(kernelParams, kernelParam)
 
+	for _, nd := range nds {
 		lkvmArgs = append(lkvmArgs, "--network")
-		lkvmArg := fmt.Sprintf("mode=tap,tapif=%s,host_ip=%s,guest_ip=%s", nd.IfName(), nd.HostIP(), nd.GuestIP())
+		lkvmArg := fmt.Sprintf("mode=tap,tapif=%s,host_ip=%s,guest_ip=%s", nd.IfName(), nd.Gateway(), nd.GuestIP())
 		lkvmArgs = append(lkvmArgs, lkvmArg)
 	}
 
-	return lkvmArgs, kernelParams, nil
+	return lkvmArgs, nil
+}
+
+func generateMacAddress() (net.HardwareAddr, error) {
+	mac := make([]byte, 6)
+	_, err := rand.Read(mac)
+	if err != nil {
+		return nil, fmt.Errorf("cannot generate random mac address: %v", err)
+	}
+
+	mac[0] = 2
+	mac[1] = 21
```

Comment:
please add a comment what 2,21,21 stands for, eg. "unicast locally administered" and you don't need to randomise all the bytes, you can `rand.Read(mac[3:5])`